File: serveur/app/models/exercise.py
# Exemple : app/models/exercise.py

from sqlalchemy import Column, Integer, String, ForeignKey, Boolean, DateTime, UniqueConstraint, func
from sqlalchemy.dialects.postgresql import JSONB
from app.database import Base

import uuid

class Exercise(Base):
    __tablename__ = "exercises"
    __table_args__ = (
        UniqueConstraint("user_id", "name", name="uq_exercises_user_id_name"),
    )

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)

    uuid = Column(String, default=lambda: str(uuid.uuid4()), nullable=False, unique=True)

    name = Column(String, nullable=False)
    description = Column(String, nullable=True)
    instructions = Column(JSONB, nullable=True)
    recommended_sets = Column(Integer, nullable=True)
    recommended_reps = Column(String, nullable=True)
    duration_in_seconds = Column(Integer, nullable=True)
    rest_time_seconds = Column(Integer, nullable=True)
    gif_url = Column(String, nullable=True)
    is_favorite = Column(Boolean, default=False)
    last_done = Column(String, nullable=True)

    updated_at = Column(DateTime(timezone=True), nullable=True)

File: appli-android/app/src/main/java/com/example/sportapp/core/data/model/Exercise.kt
package com.example.sportapp.core.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.ColumnInfo
import com.example.sportapp.core.utils.CustomDateUtils.getNowISO8601

@Entity(
    tableName = "exercises",
    indices = [androidx.room.Index(value = ["uuid"], unique = true)]
)
data class Exercise(
    @PrimaryKey
    @ColumnInfo(name = "uuid") val uuid: String,

    @ColumnInfo(name = "user_id") val userId: Int,
    val name: String,

    val description: String? = null,
    @ColumnInfo(name = "instructions") val instructions: List<String>? = null,

    @ColumnInfo(name = "recommended_sets") val recommendedSets: Int? = null,
    @ColumnInfo(name = "recommended_reps") val recommendedReps: String? = null,
    @ColumnInfo(name = "duration_in_seconds") val durationInSeconds: Int? = null,
    @ColumnInfo(name = "rest_time_seconds") val restTimeSeconds: Int? = null,
    @ColumnInfo(name = "gif_url") val gifUrl: String? = null,
    @ColumnInfo(name = "is_favorite", defaultValue = "0") val isFavorite: Boolean = false,
    @ColumnInfo(name = "last_done") val lastDone: String? = null,

    @ColumnInfo(defaultValue = "0") val synced: Boolean = false,
    @ColumnInfo(defaultValue = "0") val pendingDeletion: Boolean = false,

    @ColumnInfo(name = "updated_at") val updatedAt: String? = getNowISO8601(),
)
